diffrax/_solver/srk.py

```diff
@@ -0,0 +1,740 @@
+
+        assert np.allclose(sum(self.b_sol), 1.0)
+
+        assert self.cfs_bm.check() == self.b_sol.shape[0]
+
+        if self.b_error is not None and (not self.additive_noise):
+            assert self.cfs_bm.has_error_estimate
+
+        if self.ignore_stage_f is not None:
+            assert len(self.ignore_stage_f) == len(self.b_sol)
+        if self.ignore_stage_g is not None:
+            assert len(self.ignore_stage_g) == len(self.b_sol)
+
+
+StochasticButcherTableau.__init__.__doc__ = """**Arguments:**
+
+Let `s` denote the number of stages of the solver.
+
+- `s`: The number of stages of the solver.
+- `a`: The lower triangle (without the diagonal) of the Butcher tableau. Should
+    be a tuple of NumPy arrays, corresponding to the rows of this lower triangle. The
+    first array should be of shape `(1,)`. Each subsequent array should
+    be of shape `(2,)`, `(3,)` etc. The final array should have shape `(s - 1,)`.
+- `b_sol`: The linear combination of stages to take to produce the output at each step.
+    Should be a NumPy array of shape `(s,)`.
+- `b_error`: The linear combination of stages to take to produce the error estimate at
+    each step. Should be a NumPy array of shape `(s,)`. Note that this is *not*
+    differenced against `b_sol` prior to evaluation. (i.e. `b_error` gives the linear
+    combination for producing the error estimate directly, not for producing some
+    alternate solution that is compared against the main solution).
+- `c`: The time increments used in the Butcher tableau.
+    Should be a NumPy array of shape `(s-1,)`, as the first stage has time increment 0.
+- `cfs_bm`: An instance of a subclass of `AbstractStochasticTableau` representing
+    the coefficients for the Brownian increment and possibly its Levy areas.
+"""
+
+
+class AbstractSRK(AbstractSolver[_SolverState]):
+    r"""A general Stochastic Runge-Kutta method.
+
+    The second term in the MultiTerm must be a `ControlTerm` with
+    `control=VirtualBrownianTree`. Depending on the Butcher tableau, the
+    `VirtualBrownianTree` may need to be initialised with 'levy_area="space-time"'
+    or 'levy_area="space-time-time"'.
+
+    Given the Stratonovich SDE
+    $dy(t) = f(t, y(t)) dt + g(t, y(t)) \circ dw(t)$
+
+    We construct the SRK with $s$ stages as follows:
+
+    $y_{n+1} = y_n + h \Big(\sum_{j=1}^s b_j f_j \Big)
+    + W_n \Big(\sum_{i=1}^{j-1} b^W_j g_i \Big)
+    + H_n \Big(\sum_{i=1}^{j-1} b^H_j g_i \Big)$
+
+    $f_j = f(t_0 + c_j h , z_j)$
+
+    $g_j = g(t_0 + c_j h , z_j)$
+
+    $z_j = y_n + h \Big(\sum_{i=1}^{j-1} a_{j,i} f_i \Big)
+    + W_n \Big(\sum_{i=1}^{j-1} a^W_{j,i} g_i \Big)
+    + H_n \Big(\sum_{i=1}^{j-1} a^H_{j,i} g_i \Big)$
+
+    where $W_n = W_{t_n, t_{n+1}}$ is the increment of the Brownian motion and
+    $H_n = H_{t_n, t_{n+1}}$ is its corresponding space-time Lévy Area.
+    A similar term can also be added for the space-time-time Lévy area, K.
```

Comment:
Can you define STLA?

Reply:
(I should add -- you can see that a great many of my comments are about documentation! This is good, I think it means the code is very nearly done :)
For the docs themselves, my goal with the Diffrax docmentation is to make it so that an end user should not need to go and look at any papers -- they should be able to fairly straightforwardly figure out which solvers they would like to try without being an expert in numerical methods.
This is quite important because of the way Diffrax's interface works: you have to explicitly pick a solver and pass it in. There's very little hand-holding in the API, and the reason that this works is, I think, because the docs do their best to try and guide you through as much as possible.)

Reply:
Sounds good, I'll try to add a bit more detail.